File: week2-math/problem01-팩토리얼/heejae.cpp
// 문제: 팩토리얼
// 접근 방식: 제한 사항에 n이 3,628,800 이하라해서.. 10부터 1씩 감소하면서 팩토리얼을 구한 후 n이하라면 return한다.
// 시간복잡도: O(1)

#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    
    // n이하의 최대의 팩토리얼
    for(int i=10; i>=1; i--) {
        int temp = 1;
        
        for(int j=1; j<=i; j++) {
            temp *= j;
        }
        
        if(temp <=n) return i;
    }
    
    return answer;
}

File: week2-math/problem01-팩토리얼/heejae.java
// 문제: 팩토리얼
// 접근 방식: 입출력 제한사항이 0 < n ≤ 3,628,800 (=10 팩토리얼) 이므로..
// 10에서부터 1까지 각 숫자의 팩토리얼을 구하여 n 이하가 되는 순간 해당 숫자를 return한다.
// 시간복잡도: O(1)

class Solution {
    public int solution(int n) {
        
        for(int i=10; i>=1; i--) {
            int num = i;
            for(int j=i-1; j>=1; j--) {
                num *= j;
            }
            
            if(num > n) continue;
            else return i;
        }
        
        return 0;
    }
}
